cypress/test/lib/api.cy.js

```diff
@@ -0,0 +1,41 @@
+import { getAllPosts, getPostBySlug, getDocBySlug } from '../../../lib/api';
+import posts from '../../../config/posts.json'
+describe('getAllPosts', () => {
+  it('should return all posts', () => {
+    const allPosts = getAllPosts();
+    expect(allPosts).to.deep.equal(posts);
+  });
+});
+
+describe('getPostBySlug', () => {
+  it('should return the post with the given slug', () => {
+    const slug = '/blog/2023-may-docs-report';
+    const post = getPostBySlug(slug);
+    const expectedPost = posts.blog.find((p) => p.slug === slug && !p.isSection);
+    expect(post).to.deep.equal(expectedPost);
+  });
+
+  it('should return the post of a specific type with the given slug', () => {
+    const slug = '/blog/2023-may-docs-report';
+    const type = 'blog';
+    const post = getPostBySlug(slug, type);
+    const expectedPost = posts[type].find((p) => p.slug === slug && !p.isSection);
+    expect(post).to.deep.equal(expectedPost);
+  });
+});
+
+describe('getDocBySlug', () => {
+    it('should return the document with the given slug', () => {
+      const structuredPosts = [
+        { slug: 'doc1', isSection: false },
+        { slug: 'doc2', isSection: false },
+        { slug: 'doc3', isSection: false }
+      ];
```

Comment:
Use `posts['docs']` for the structured posts and test the function same as above.

Reply:
done